pkg/logs/launchers/integration/launcher.go

```diff
@@ -0,0 +1,202 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package integration creates a launcher to track logs from integrations
+package integration
+
+import (
+	"os"
+	"strings"
+
+	ddLog "github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/datadog-agent/comp/logs/agent/config"
+	"github.com/DataDog/datadog-agent/comp/logs/integrations/def"
+	pkgConfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/logs/auditor"
+	"github.com/DataDog/datadog-agent/pkg/logs/launchers"
+	"github.com/DataDog/datadog-agent/pkg/logs/pipeline"
+	"github.com/DataDog/datadog-agent/pkg/logs/sources"
+	"github.com/DataDog/datadog-agent/pkg/logs/tailers"
+)
+
+// Launcher checks for launcher integrations, creates files for integrations to
+// write logs to, then creates file sources for the file launcher to tail
+type Launcher struct {
+	sources              *sources.LogSources
+	addedSources         chan *sources.LogSource
+	stop                 chan struct{}
+	done                 chan struct{}
+	runPath              string
+	integrationsLogsChan chan integrations.IntegrationLog
+	integrationToFile    map[string]string
+	// writeLogToFile is used as a function pointer so it can be overridden in
+	// testing to make deterministic tests
+	writeFunction func(filepath, log string) error
+}
+
+// NewLauncher returns a new launcher
+func NewLauncher(sources *sources.LogSources, integrationsLogsComp integrations.Component) *Launcher {
+	return &Launcher{
+		sources:              sources,
+		runPath:              pkgConfig.Datadog().GetString("logs_config.run_path"),
+		stop:                 make(chan struct{}),
+		done:                 make(chan struct{}),
+		integrationsLogsChan: integrationsLogsComp.Subscribe(),
+		integrationToFile:    make(map[string]string),
+		writeFunction:        writeLogToFile,
+	}
+}
+
+// Start starts the launcher and launches the run loop in a go function
+func (s *Launcher) Start(sourceProvider launchers.SourceProvider, _ pipeline.Provider, _ auditor.Registry, _ *tailers.TailerTracker) {
+	s.addedSources, _ = sourceProvider.SubscribeForType(config.IntegrationType)
+
+	go s.run()
+}
+
+// Stop stops the scanner
+func (s *Launcher) Stop() {
+	s.stop <- struct{}{}
+	<-s.done
+}
+
+// run checks if there are new files to tail and tails them
+func (s *Launcher) run() {
+	defer func() {
```

Comment:
NIT
```suggestion
	defer close(s.done)
```

technically - you don't need to `done` channel right now since you are no longer releasing any resources when the loop exists. Since `stop` is unbuffered, the `Stop()` function will block until the loop is terminated.

Reply:
I ended up removing the done channel since it wasn't doing anything 👍